Wasi testing (#34)
SCIInstitute/ITKCleaver: CMakeLists.txt

include(FetchContent)

> set(_itk_build_testing ${BUILD_TESTING})
> if(WASI OR EMSCRIPTEN)
>   # Cleaver lib vendored GTest incompatibility
>   set(BUILD_TESTING OFF CACHE BOOL "Build testing code" FORCE)
> endif()

Reviewer: ```suggestion
# Testing
if(DEFINED BUILD_TESTING)
  set(_itk_build_testing ${BUILD_TESTING})
else()
  set(_itk_build_testing OFF) # Default
endif()
# Since Cleaver internally adds an external project downloading googletest
# that would conflict with the one provided by ITK, we systematically prevent
# Cleaver test from being configured & built.
set(BUILD_TESTING OFF CACHE BOOL "Build testing code" FORCE)
```

Author: The Cleaver2 lib vendored GTest incompatibility is only with the wasm toolchains. For other builds, we do want to be able to build/enable the ITK tests.

Author: I incorporated the first part of this improvement:

```
if(DEFINED BUILD_TESTING)
  set(_itk_build_testing ${BUILD_TESTING})
else()
  set(_itk_build_testing OFF) # Default
endif()
```

In to the other committed suggestion.

> set(BUILD_TESTING ${_itk_build_testing})

Reviewer: ```suggestion
set(BUILD_TESTING ${_itk_build_testing} CACHE BOOL "Build testing code" FORCE)
```

  include(ITKModuleExternal)
  if(WASI OR EMSCRIPTEN)
    set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})

>     if (WASI)
>       set(BUILD_TESTING ON CACHE BOOL "Build testing code" FORCE)
>     endif()

Reviewer: ```suggestion
```
